proxy/proxy_test.go

```diff
@@ -2044,14 +2044,70 @@ var _ = Describe("Proxy", func() {
 				//since the building of the log record happens throughout the life of the request
 				b, err := os.ReadFile(f.Name())
 				Expect(err).NotTo(HaveOccurred())
-				Expect(strings.HasPrefix(string(b), "test - [")).To(BeTrue())
+				Expect(string(b)).To(HavePrefix("test - ["))
 				Expect(string(b)).To(ContainSubstring(`"POST / HTTP/1.1" 200 4 6 "-"`))
 				Expect(string(b)).To(ContainSubstring(`x_forwarded_for:"127.0.0.1" x_forwarded_proto:"http" vcap_request_id:`))
 				Expect(string(b)).To(ContainSubstring(`response_time:`))
 				Expect(string(b)).To(ContainSubstring(`app_id:"456"`))
 				Expect(string(b)).To(ContainSubstring(`app_index:"2"`))
 				Expect(b[len(b)-1]).To(Equal(byte('\n')))
 			})
+
+			It("returns the correct status code to the client", func() {
+				ln := test_util.RegisterConnHandler(r, "statusTest", func(conn *test_util.HttpConn) {
+					req, body := conn.ReadRequest()
+					Expect(req.Method).To(Equal("POST"))
+					Expect(req.URL.Path).To(Equal("/"))
+					Expect(req.ProtoMajor).To(Equal(1))
+					Expect(req.ProtoMinor).To(Equal(1))
+
+					Expect(body).To(Equal("ABCD"))
+
+					expectRsp := test_util.NewResponse(100)
+					conn.WriteResponse(expectRsp)
+
+					rsp := test_util.NewResponse(201)
+					rsp.Body = io.NopCloser(strings.NewReader("DEFGHI"))
```

Comment:
It looks like we return the value `DEFGHI` but we only check that there are non-400 status codes in the response. Should we change these strings to something obvious like `valid-unimportant-response-data`?